### PR TITLE
fix: do not update apt sources for devel images

### LINKED ISSUE #503: Core24 images failing due to changes in apt source files
### Bug Description

Core24/noble images are failing because craft-providers expects `/etc/apt/sources.list` to exist.  This file was recently moved to `/etc/apt/sources.list.d/ubuntu.sources` in the latest noble daily images.

Source: https://bugs.launchpad.net/snapcraft/+bug/2051567

### To Reproduce

Launch a noble image with LXD

### part yaml

_No response_

### Relevant log output

```shell
2024-01-29 14:23:31.662 Executing in container: lxc --project snapcraft exec local:base-instance-snapcraft-buildd-base-v60--02164e19b5c3d4e594dc -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin SNAPCRAFT_MANAGED_MODE=1 SNAPCRAFT_BUILD_INFO=1 systemctl restart systemd-resolved
2024-01-29 14:23:31.976 Executing on host: lxc --project snapcraft file push /tmp/tmpbnyxf259 local:base-instance-snapcraft-buildd-base-v60--02164e19b5c3d4e594dc/etc/systemd/network/10-eth0.network --mode=0644
2024-01-29 14:23:32.046 Executing in container: lxc --project snapcraft exec local:base-instance-snapcraft-buildd-base-v60--02164e19b5c3d4e594dc -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin SNAPCRAFT_MANAGED_MODE=1 SNAPCRAFT_BUILD_INFO=1 chown root:root /etc/systemd/network/10-eth0.network
2024-01-29 14:23:32.145 Executing in container: lxc --project snapcraft exec local:base-instance-snapcraft-buildd-base-v60--02164e19b5c3d4e594dc -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin SNAPCRAFT_MANAGED_MODE=1 SNAPCRAFT_BUILD_INFO=1 systemctl enable systemd-networkd
2024-01-29 14:23:32.749 Executing in container: lxc --project snapcraft exec local:base-instance-snapcraft-buildd-base-v60--02164e19b5c3d4e594dc -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin SNAPCRAFT_MANAGED_MODE=1 SNAPCRAFT_BUILD_INFO=1 systemctl restart systemd-networkd
2024-01-29 14:23:32.979 Waiting for networking to be ready...
2024-01-29 14:23:32.979 Executing in container: lxc --project snapcraft exec local:base-instance-snapcraft-buildd-base-v60--02164e19b5c3d4e594dc -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin SNAPCRAFT_MANAGED_MODE=1 SNAPCRAFT_BUILD_INFO=1 getent hosts snapcraft.io
2024-01-29 14:23:33.109 Executing on host: lxc --project snapcraft file push /tmp/tmp3orstlip local:base-instance-snapcraft-buildd-base-v60--02164e19b5c3d4e594dc/etc/apt/apt.conf.d/00no-recommends --mode=0644
2024-01-29 14:23:33.179 Executing in container: lxc --project snapcraft exec local:base-instance-snapcraft-buildd-base-v60--02164e19b5c3d4e594dc -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin SNAPCRAFT_MANAGED_MODE=1 SNAPCRAFT_BUILD_INFO=1 chown root:root /etc/apt/apt.conf.d/00no-recommends
2024-01-29 14:23:33.282 Executing on host: lxc --project snapcraft file push /tmp/tmps_bynmr6 local:base-instance-snapcraft-buildd-base-v60--02164e19b5c3d4e594dc/etc/apt/apt.conf.d/00update-errors --mode=0644
2024-01-29 14:23:33.350 Executing in container: lxc --project snapcraft exec local:base-instance-snapcraft-buildd-base-v60--02164e19b5c3d4e594dc -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin SNAPCRAFT_MANAGED_MODE=1 SNAPCRAFT_BUILD_INFO=1 chown root:root /etc/apt/apt.conf.d/00update-errors
2024-01-29 14:23:33.453 Executing in container: lxc --project snapcraft exec local:base-instance-snapcraft-buildd-base-v60--02164e19b5c3d4e594dc -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin SNAPCRAFT_MANAGED_MODE=1 SNAPCRAFT_BUILD_INFO=1 cat /etc/os-release
2024-01-29 14:23:33.553 Updating apt sources from 'noble' to 'devel'.
2024-01-29 14:23:33.554 Executing in container: lxc --project snapcraft exec local:base-instance-snapcraft-buildd-base-v60--02164e19b5c3d4e594dc -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin SNAPCRAFT_MANAGED_MODE=1 SNAPCRAFT_BUILD_INFO=1 sed -i s/noble/devel/g /etc/apt/sources.list
2024-01-29 14:23:33.654 craft-providers error: Failed to update '/etc/apt/sources.list'.
* Command that failed: 'lxc --project snapcraft exec local:base-instance-snapcraft-buildd-base-v60--02164e19b5c3d4e594dc -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin SNAPCRAFT_MANAGED_MODE=1 SNAPCRAFT_BUILD_INFO=1 sed -i s/noble/devel/g /etc/apt/sources.list'
* Command exit code: 2
* Command standard error output: b"sed: can't read /etc/apt/sources.list: No such file or directory\n"
2024-01-29 14:23:33.654 Full execution log: '/root/.local/state/snapcraft/log/snapcraft-20240129-142232.221918.log'
```

### PR DESCRIPTION
**Pull request by mr-cal**
- [x] Have you followed the guidelines for contributing?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `tox`?

-----

Fixes #503

Additionally, this is a workaround for https://bugs.launchpad.net/ubuntu/+source/apt/+bug/1728616

### CODE REVIEW

File: craft_providers/bases/ubuntu.py [hunk 1]
```diff
@@ -177,98 +177,6 @@ def _ensure_os_compatible(self, executor: Executor) -> None:
...
-        except subprocess.CalledProcessError as error:
-            raise BaseConfigurationError(
-                brief=(
-                    f"Could not update {cloud_config!r} because it is empty or "
-                    "does not exist."
-                ),
-                details=details_from_called_process_error(error),
-            ) from error
-
-        # update cloud.cfg to prevent the sources.list file from being reset
```
Comment (tigarmo) [inline craft_providers/bases/ubuntu.py:224]: so I understand that the motivation for this PR is that the noble images moved the sources file, but how come we don't have to do all of this anymore? As opposed to updating the code to use the new file?

Comment (mr-cal) [reply craft_providers/bases/ubuntu.py:224]: We can update source files in the ubuntu daily devel image, but we run into [this bug](https://bugs.launchpad.net/ubuntu/+source/apt/+bug/1728616) when we try to update source files in the buildd daily devel image.

Why does that bug only affect buildd daily images and not ubuntu daily images?  No idea!

Last week, @sergiusens said we could stop updating the source files in devel images as long as we expire the base instances every 14 days (which is [already implemented](https://github.com/canonical/craft-providers/blob/19c6a2baf2fd1cc8b5706158c6ed5ebc79fe9363/craft_providers/lxd/lxd_provider.py#L149))

Comment (syu-w) [reply craft_providers/bases/ubuntu.py:224]: So for regular images, we don't need to update the source files?

Comment (mr-cal) [reply craft_providers/bases/ubuntu.py:224]: Correct, this was only for [`devel` images](https://github.com/canonical/craft-providers/pull/504/files#diff-c042ac17cabf0d18e3301f2f93aede60dd3a57f6952f3ee1bac6b48a3d62fa03L297)

**Review by syu-w — APPROVED** [review]

File: craft_providers/bases/ubuntu.py (hunk 1 above)
Comment (tigarmo) [reply craft_providers/bases/ubuntu.py:224]: ah I see; the change that I skipped over was the removal of the `if self.alias == BuilddBaseAlias.DEVEL:` code a bit further down on the file.

**Review by tigarmo — APPROVED** [review]

**Comment by mr-cal** [comment]
Only failures in this PR are for almalinux, which is a known problem (#502)

**Pull request merged**